### PR TITLE
Polish the SiteRecovery for stable release

### PR DESCRIPTION
**Pull request by ArthurMa1978**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 1]
```diff
@@ -5090,161 +4199,1198 @@ public partial class ResourceHealthSummary
...
+    }
+    public partial class SiteRecoveryVmDiskDetails
+    {
+        internal SiteRecoveryVmDiskDetails() { }
+        public string CustomTargetDiskName { get { throw null; } }
+        public Azure.Core.ResourceIdentifier DiskEncryptionSetId { get { throw null; } }
+        public string DiskId { get { throw null; } }
+        public string LunId { get { throw null; } }
+        public string MaxSizeMB { get { throw null; } }
+        public string TargetDiskLocation { get { throw null; } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5336]: Is this AzureLocation?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 2]
```diff
@@ -5923,22 +5923,22 @@ public partial class VMwareCbtProtectionContainerMappingDetails : Azure.Resource
...
         public System.Collections.Generic.IReadOnlyList<string> ExcludedSkus { get { throw null; } }
-        public string KeyVaultId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier KeyVaultId { get { throw null; } }
         public System.Uri KeyVaultUri { get { throw null; } }
         public System.Collections.Generic.IReadOnlyDictionary<string, int> RoleSizeToNicCountMap { get { throw null; } }
         public string ServiceBusConnectionStringSecretName { get { throw null; } }
-        public string StorageAccountId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier StorageAccountId { get { throw null; } }
         public string StorageAccountSasSecretName { get { throw null; } }
         public string TargetLocation { get { throw null; } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5932]: Is this AzureLocation?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 3]
```diff
@@ -5833,63 +5833,63 @@ public partial class VMwareCbtMigrationDetails : Azure.ResourceManager.RecoveryS
...
+        public Azure.Core.ResourceIdentifier StorageAccountId { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<string> SupportedOSVersions { get { throw null; } }
-        public string TargetAvailabilitySetId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier TargetAvailabilitySetId { get { throw null; } }
         public string TargetAvailabilityZone { get { throw null; } }
-        public string TargetBootDiagnosticsStorageAccountId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier TargetBootDiagnosticsStorageAccountId { get { throw null; } }
         public System.Collections.Generic.IReadOnlyDictionary<string, string> TargetDiskTags { get { throw null; } }
         public string TargetGeneration { get { throw null; } }
         public string TargetLocation { get { throw null; } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5847]: Could this be AzureLocation?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 4]
```diff
@@ -6023,36 +6023,36 @@ public partial class VMwareDetails : Azure.ResourceManager.RecoveryServicesSiteR
...
         internal VMwareV2FabricSpecificDetails() { }
-        public string MigrationSolutionId { get { throw null; } }
-        public string PhysicalSiteId { get { throw null; } }
-        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.ProcessServerDetails> ProcessServers { get { throw null; } }
+        public Azure.Core.ResourceIdentifier MigrationSolutionId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier PhysicalSiteId { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.SiteRecoveryProcessServerDetails> ProcessServers { get { throw null; } }
         public string ServiceContainerId { get { throw null; } }
         public string ServiceEndpoint { get { throw null; } }
         public string ServiceResourceId { get { throw null; } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:6041]: Is this arm-id?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 5]
```diff
@@ -5833,63 +5833,63 @@ public partial class VMwareCbtMigrationDetails : Azure.ResourceManager.RecoveryS
...
+        public Azure.Core.ResourceIdentifier TargetProximityPlacementGroupId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier TargetResourceGroupId { get { throw null; } }
         public string TargetVmName { get { throw null; } }
         public Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.VMwareCbtSecurityProfileProperties TargetVmSecurityProfile { get { throw null; } }
         public string TargetVmSize { get { throw null; } }
         public System.Collections.Generic.IReadOnlyDictionary<string, string> TargetVmTags { get { throw null; } }
-        public string TestNetworkId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier TestNetworkId { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.VMwareCbtNicDetails> VmNics { get { throw null; } }
         public string VMwareMachineId { get { throw null; } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5858]: Is this arm-id?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 6]
```diff
@@ -6023,36 +6023,36 @@ public partial class VMwareDetails : Azure.ResourceManager.RecoveryServicesSiteR
...
     public partial class VMwareV2FabricSpecificDetails : Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.FabricSpecificDetails
     {
         internal VMwareV2FabricSpecificDetails() { }
-        public string MigrationSolutionId { get { throw null; } }
-        public string PhysicalSiteId { get { throw null; } }
-        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.ProcessServerDetails> ProcessServers { get { throw null; } }
+        public Azure.Core.ResourceIdentifier MigrationSolutionId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier PhysicalSiteId { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.SiteRecoveryProcessServerDetails> ProcessServers { get { throw null; } }
         public string ServiceContainerId { get { throw null; } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:6039]: Is this arm-id?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 7]
```diff
@@ -5090,161 +4199,1198 @@ public partial class ResourceHealthSummary
...
+        public System.Collections.Generic.IList<string> CustomEmailAddresses { get { throw null; } }
+        public string Locale { get { throw null; } set { } }
+        public string SendToOwners { get { throw null; } set { } }
+    }
+    public partial class SiteRecoveryCreateProtectionContainerMappingProperties
+    {
+        public SiteRecoveryCreateProtectionContainerMappingProperties() { }
+        public Azure.Core.ResourceIdentifier PolicyId { get { throw null; } set { } }
+        public Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.ReplicationProviderSpecificContainerMappingContent ProviderSpecificContent { get { throw null; } set { } }
+        public string TargetProtectionContainerId { get { throw null; } set { } }
```
Comment (live1206) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:4414]: Is this arm-id?

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/autorest.md [hunk 8]
```diff
@@ -15,18 +15,315 @@ skip-csproj: true
 modelerfour:
   flatten-payloads: false
 
+#mgmt-debug: 
+#  show-serialized-names: true
+
 rename-mapping:
+  Alert: SiteRecoveryAlert
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/autorest.md:22]: RP prefix is `RecoveryServicesSiteRecovery` which is too long

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 9]
```diff
@@ -816,276 +280,823 @@ public partial class ReplicationProtectionIntentResource : Azure.ResourceManager
...
-        public string RecoveryCapacityReservationGroupId { get { throw null; } set { } }
+        public Azure.Core.ResourceIdentifier RecoveryAzureNetworkId { get { throw null; } set { } }
+        public Azure.Core.ResourceIdentifier RecoveryBootDiagStorageAccountId { get { throw null; } set { } }
+        public Azure.Core.ResourceIdentifier RecoveryCapacityReservationGroupId { get { throw null; } set { } }
         public string RecoveryCloudServiceId { get { throw null; } set { } }
-        public string RecoveryContainerId { get { throw null; } set { } }
-        public Azure.ResourceManager.RecoveryServicesSiteRecovery.Models.ExtendedLocation RecoveryExtendedLocation { get { throw null; } set { } }
-        public string RecoveryProximityPlacementGroupId { get { throw null; } set { } }
-        public string RecoveryResourceGroupId { get { throw null; } set { } }
+        public Azure.Core.ResourceIdentifier RecoveryContainerId { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:1050]: It's not arm-id

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs [hunk 10]
```diff
@@ -3584,147 +3231,147 @@ public partial class InMageRcmReplicationDetails : Azure.ResourceManager.Recover
...
         public string RunAsAccountId { get { throw null; } }
-        public string StorageAccountId { get { throw null; } }
-        public string TargetAvailabilitySetId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier StorageAccountId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier TargetAvailabilitySetId { get { throw null; } }
         public string TargetAvailabilityZone { get { throw null; } }
-        public string TargetBootDiagnosticsStorageAccountId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier TargetBootDiagnosticsStorageAccountId { get { throw null; } }
         public string TargetGeneration { get { throw null; } }
         public string TargetLocation { get { throw null; } }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:3242]: Unable to confirm if it's `AzureLocation` or not, leave it as is.

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5336]: Unable to confirm if it's `AzureLocation` or not, leave it as is.

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5932]: Unable to confirm if it's `AzureLocation` or not, leave it as is.

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5847]: Unable to confirm if it's `AzureLocation` or not, leave it as is.

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 7 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:4414]: fixed

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:6039]: fixed

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 5 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:5858]: fixed

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs (hunk 4 above)
Comment (ArthurMa1978) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/api/Azure.ResourceManager.RecoveryServicesSiteRecovery.netstandard2.0.cs:6041]: fixed

**Review by live1206 — APPROVED** [review]

**Pull request merged**